prune.go

```diff
@@ -134,8 +134,9 @@ func maybePrune(db *mgo.Database, txnsName string, pruneOpts PruneOptions) error
 	}
 
 	stats, err := CleanAndPrune(CleanAndPruneArgs{
-		Txns:      txns,
-		TxnsCount: txnsCount,
+		Txns:       txns,
+		TxnsCount:  txnsCount,
+		NewestTime: pruneOpts.NewestTime,
```

Comment:
So all this does is allow the caller to specify a time threshold.

One thing that could be done here as part of 'maybePrune' is to default to passing
lastPrune.Started instead of just doing pass through.
Then we would default to whatever window that we trigger pruning on, without callers having to track it.
mgopurge would have to track something, as it doesn't use maybePrune.

For now, I can just update callers to do:
NewestTime: time.Now().Add(-time.Hour)

FWIW, I'm also not very happy with "NewestTime", I just didn't have a better word.